include/arch/arm/arch.h

```diff
@@ -42,9 +42,98 @@ extern "C" {
 #include <arch/arm/cortex_m/sys_io.h>
 #include <arch/arm/cortex_m/nmi.h>
 #endif
+#if defined(CONFIG_MPU_STACK_GUARD)
+
```

Comment:
Sorry for the late comment, but I found out that we need this regardless of whether or not the STACK_GUARD is set.  The reason being that we need these in place to place the stacks in a specific place.  Instead of noinit, we can set the location to a specific region during link.

How about just putting the STACK_SIZE in the #if def and set it to 32 or 4 based on the STACK_GUARD being set.  The rest would always be defined.